### PR TITLE
Make deleteObject() and deleteBucket() return false on 404 response

### PR DESCRIPTION
**Pull request by snotling**

### CODE REVIEW

File: google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/features/ObjectApi.java [hunk 1]
```diff
@@ -199,7 +198,7 @@ GoogleCloudStorageObject simpleUpload(@PathParam("bucket") String bucketName, @H
    @Named("Object:delete")
    @DELETE
    @Path("storage/v1/b/{bucket}/o/{object}")
-   @Fallback(TrueOnNotFoundOr404.class)
+   @Fallback(FalseOnNotFoundOr404.class)
```
Comment (andrewgaul) [inline google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/features/ObjectApi.java]: Which test exercises this functionality?

Comment (snotling) [inline google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/features/ObjectApi.java]: In all the tests for GCS, the return value of ``deleteObject()`` calls is not checked.
So I think the tests should be improved in that regard.
Is that what you meant?

Comment (andrewgaul) [inline google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/features/ObjectApi.java]: Definitely, can you either modify an existing test or add a new test?  We need not check at all call sites, but at least the basic delete test should do this.  Thanks!

Comment (snotling) [inline google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/features/ObjectApi.java]: Is the current ``master`` GCS test suite supposed to pass? I'm having a few errors, in particular in ``ObjectApiLiveTest`` (the class I'm working on).

Comment (andrewgaul) [inline google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/features/ObjectApi.java]: @snotling I opened [JCLOUDS-805](https://issues.apache.org/jira/browse/JCLOUDS-805) to track this issue.  I will take a quick look and asked @hsbhathiya, the original developer, to do the same.

**Comment by danbroudy** [comment]
Im adding ObjectApiMockTest because there doesn't appear to be a unit test and it would be good to convert GCS to Mock

File: google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/features/ObjectApi.java (hunk 1 above)
Comment (andrewgaul) [inline google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/features/ObjectApi.java]: @snotling The GCS tests now pass without error; can you rebase your commit and revisit this pull request with new tests?

Comment (snotling) [inline google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/features/ObjectApi.java]: I've extended the "return-false-on-404" behavior to `deleteBucket()` as well.
And added a couple of unit tests.

File: google-cloud-storage/src/test/java/org/jclouds/googlecloudstorage/features/ObjectApiLiveTest.java [hunk 2]
```diff
@@ -422,8 +425,9 @@ public void testDeleteObjectWithOptions() {
 
    @AfterClass
    private void deleteBucket() {
-      api.getBucketApi().deleteBucket(BUCKET_NAME);
-      api.getBucketApi().deleteBucket(BUCKET_NAME2);
+      assertTrue(api.getBucketApi().deleteBucket(BUCKET_NAME));
+      assertTrue(api.getBucketApi().deleteBucket(BUCKET_NAME2));
+      assertFalse(api.getBucketApi().deleteBucket(NONEXISTENT_BUCKET_NAME));
```
Comment (andrewgaul) [inline google-cloud-storage/src/test/java/org/jclouds/googlecloudstorage/features/ObjectApiLiveTest.java]: Should `@AfterClass` have asserts?  This method should just clean up after the test mehod.  Instead can you create a separate `testDeleteBucket` method?

Comment (snotling) [inline google-cloud-storage/src/test/java/org/jclouds/googlecloudstorage/features/ObjectApiLiveTest.java]: Indeed, having a dedicated test method is more appropriate.
I'll do that.

Comment (hsbhathiya) [inline google-cloud-storage/src/test/java/org/jclouds/googlecloudstorage/features/ObjectApiLiveTest.java]: I think just removing asserts would be enough. Delete bucket tests are available in BucketApiLiveTests

**Comment by snotling** [comment]
All tests now pass.
By the way, am I overlooking something or Jenkins only shows the overall build status and not the actual errors when the integration tests fail?

**Comment by andrewgaul** [comment]
@snotling Jenkins only runs unit tests for us.  Unfortunately integration tests ar a manual process due to various automation issues.

**Comment by andrewgaul** [comment]
Ran integration tests locally, squashed commits, and pushed to master as 6f9783ce4c316d93d6eac209967cc0a86152844f and 1.9.x as cbb6e88294696225da7e97f40e72aa26239c2651.

**Pull request closed without merging**